### ISSUE TITLE
Review code

### ISSUE THREAD

**Issue by mhoff** [team-pltb/pltb#2]
Check complete code and fix remaining warnings.

**Comment by mhoff** [comment]
@briehm Is here happening something?
I still see warnings when compiling our code.

**Comment by mhoff** [comment]
Push

**Comment by briehm** [comment]
> src/mpi_masterworker.c: In function 'worker':
> src/mpi_masterworker.c:173:24: warning: unused parameter 'process_id' [-Wunused-parameter]
> static void worker(int process_id, int master_id,

It seems like GCC is not getting that the process_id is used inside the Debug-Macros:

> DBG_WORKER("Worker[%02d]: Received order to process matrix #%u\n",
>                   process_id, task.matrix_index);

The Clang compiler warns about either GNU Extensions and Macros or the remaining (afaik unfixable) Padding issues..

**Comment by mhoff** [comment]
> It seems like GCC is not getting that the process_id is used inside the Debug-Macros

Yes. That's because the preprocessor eliminates those before the compiler does its actual work.
I think the only solution would be to make also the method header itself dependent on macros.
In debug mode with the additional `int` parameter and without it otherwise.

But I think that would be overcomplicated. What do you mean?

> The Clang compiler warns about either GNU Extensions and Macros or the remaining (afaik unfixable) Padding issues..

Yeah. I think thats okay.
